### PR TITLE
Updated Readme file with installation and usage instructions

### LINKED ISSUE #1: Update README file with setup instructions and usage
Add a Readme file with instructions on installation, usage, development, and building and releasing.

@AbhayVAshokan _a

### PR DESCRIPTION
**Pull request by AbhayVAshokan**
Fixes #1 

- Added `README.md` file with instructions for installation, usage, development, and building.

@Amaljith-K _a please have a look.

### CODE REVIEW

**Review by Amaljith-K — COMMENTED** [review]
LGTM except for these comments

**Comment by AbhayVAshokan** [comment]
@Amaljith-K _a please have a look. I've replaced the `data` method with `process`. I've seen this usage in kb and chat.

**Comment by Amaljith-K** [comment]
LGTM
@unnitallman _a WDYT?

**Comment by unnitallman** [comment]
@AbhayVAshokan What will `params[:filters]`  be like? That needs to be in the doc as well.

File: README.md [hunk 1]
```diff
@@ -1 +1,50 @@
...
+  end
+
+private
+
+  def load_users
+    @users = @organization.users
+  end
+
+  def filtered_users
+    @filter_service ||= NeetoFilters::FilterService.new(@users, params[:filters])
```
Comment (unnitallman) [inline README.md:45]: @AbhayVAshokan https://vimeo.com/693400163/99e37af8c0

cc: @Amaljith-K

Comment (Amaljith-K) [reply README.md:45]: @unnitallman good suggestion.
Here is an overall picture of how the gem should behave: https://vimeo.com/693480672/4f4a81d84a
@AbhayVAshokan @unnitallman let me know your opinions on this.

Comment (unnitallman) [reply README.md:45]: Looks good to me

**Comment by AbhayVAshokan** [comment]
@Amaljith-K _a can you please add a section about custom attributes and additional setup required on the project side so that it would work as intended in the services?

**Comment by Amaljith-K** [comment]
Since `dynamic_search_services` is a subset of `custom_search_services`, let us combined them both in `custom_filter_services`
https://vimeo.com/694340552/85745bf71c

@AbhayVAshokan I've updated the PR with documentation for custom filtration service configuration.

@unnitallman _a please review the changes in: https://github.com/bigbinary/neeto-filters-backend/pull/2/commits/61e4d1f71cdecc88232bc8362e2b05f13adc9ccf

**Pull request merged**